### DISCUSSION TITLE
update from `cloudsql postgres 9.6` to `cloudsql postgres 12`

### DISCUSSION THREAD

**Discussion by gowrisankar22** [concourse/concourse#5718] [General]
Our concourse setup is running on GCP and we are cloudsql is our dB and we using quite old version 9.6. We are planning to upgrade from `cloudsql postgres 9.6` to `cloudsql postgres 12` 

Was this update recommended and tested or out of the box it will just work by just changing the version in the terraform ? 

I can see similar thing in https://github.com/concourse/hush-house/issues/24

**Comment by gowrisankar22** [comment]
@vito @jamieklassen Any suggestions on this ?

**Comment by gowrisankar22** [comment]
@jamieklassen any update on this?

**Reply by jamieklassen** [reply]
no. we have no plans to try it ourselves any time soon -- if you have a staging environment, give it a try!

**Reply by gowrisankar22** [reply]
@jamieklassen Sure I can try this out but question is Work load wise my staging system(10pipelines) has Very less load compared to prod (500 pipelines) . Does this matter ? If the migration works fine in staging , will this work fine in prod also ?(different workload won’t cause any issue?)

**Reply by jamieklassen** [reply]
dunno! i really don't know what changes to expect between postgres 9 and 12

**Reply by gowrisankar22** [reply]
Ah ok.. let me give it a try and see in staging.. thanks @jamieklassen

**Reply by gowrisankar22** [reply]
@jamieklassen I managed to do this upgrade without any issues. thanks.

**Reply by jamieklassen** [reply]
@gowrisankar22 sorry I just realized that our CI pipelines have been running basic integration tests against the latest `postgres` image on dockerhub, which is 12.3: https://hub.docker.com/layers/postgres/library/postgres/latest/images/sha256-222c12ec1601f38cee2c84fd044756e32815cf11ce36151735c37cb4ead60855?context=explore

**Comment by gareth-e** [comment]
Hi @gowrisankar22 . We're about to start a similar upgrade at our site. We have a similar setup / scale to you (maybe a few less pipelines). Concourse version is `6.7.1` for us.
Did you have any issues? Or do you have any good tips or recommendations from your upgrade last year?

**Reply by ibmer921150** [reply]
@gowrisankar22  @ch-gareth, hey guys, our team would to do the PostgresSQL update from (9.6) to (12). I am also keen to know if any breaking changes for concourse during the PostgresSQL upgrade. Thanks.

**Reply by taylorsilva** [reply]
Hey folks, there shouldn't be any breaking changes. We don't test this postgres upgrade path, but we do run tests on portgres 9 and postgres 12. Concourse works on both. If you're just updating postgres and not Concourse at the same time then I'd personally expect nothing to break on the Concourse side. There will be downtime I'm guessing since postgres will have to restart, but the web nodes should continue to work as usual.

**Reply by ibmer921150** [reply]
Hi @taylorsilva, thanks for your reply. As our Concourse version is pretty old V5.0.1. Can you please let me know if the old Concourse version also works on ProstgreSQL 12? It would be nice to see the Concourse version and PostgreSQL version compatibility. Thanks

**Reply by taylorsilva** [reply]
I have no way of 100% knowing without doing a bunch of testing, but I will say that I'm highly confident that v5 of Concourse would work fine on Postgres 12.
